include/uk/event.h

```diff
@@ -251,24 +253,32 @@ struct uk_event {
 
 /**
  * Raise an event by pointer and invoke the handler chain until the first
- * handler successfully handled the event.
+ * handler successfully handled the event or returned a negative error code.
  *
- * @param event
+ * @param e
  *   Pointer to the event to raise.
  * @param data
  *   Optional data supplied to the event handlers
- * @return
- *   One of the UK_EVENT_* macros on success, errno on < 0
+ * @returns
+ *   UK_EVENT_HANDLED if at least one handler indicated that it successfully
+ *   handled the event, otherwise returns UK_EVENT_NOT_HANDLED. If a handler
+ *   returns a negative value, then this negative value is returned.
  */
 static inline int uk_raise_event_ptr(struct uk_event *e, void *data)
 {
 	const uk_event_handler_t *itr;
+	int rc;
 	int ret = UK_EVENT_NOT_HANDLED;
 
 	uk_event_handler_foreach(itr, e) {
 		__uk_event_assert(*itr);
-		ret |= ((*itr)(data));
-		if (ret == UK_EVENT_HANDLED)
+		rc = ((*itr)(data));
+		if (rc < 0)
+			return rc;
```

Comment:
I won't get tired saying this. Can you please add the `unlikely` for error cases 😛